### PR TITLE
Spark Dotnet: fix API links

### PR DESCRIPTION
**Pull request by mamccrea**
## Summary

Changing API links.

### CODE REVIEW

**Comment by mairaw** [comment]
You only have that one namespace @mamccrea? It seems that your repo wasn't indexed given that https://docs.microsoft.com/en-us/dotnet/api/?view=spark-dotnet is empty.

**Comment by JasonWHowell** [comment]
@mamccrea looks like the index includes .Net for Spark now. 
image

**Comment by JasonWHowell** [comment]
@mamccrea Do you want the new link going to the Microsoft.Spark namespace directly, or to the index of all the namespaces in the listing (original link)?

**Comment by mamccrea** [comment]
@JasonWHowell - I see the index is working now, but for some reason I'm getting a 404 when I try to get there from the docs index page (https://docs.microsoft.com/en-us/dotnet/spark/). I'll investigate that.

**Comment by mairaw** [comment]
That one is missing the dotnet in the path.

File: docs/spark/index.yml [hunk 1]
```diff
@@ -29,5 +29,5 @@ sections:
   - type: list
     style: unordered
     items:
-    - html: '<a href="/api/?view=spark-dotnet">.NET for Apache Spark API</a>'
+    - html: '<a href="https://docs.microsoft.com/dotnet/api/Microsoft.Spark">.NET for Apache Spark API</a>'
```
Comment (mairaw) [inline docs/spark/index.yml]: This should work

```suggestion
    - html: '<a href="/dotnet/api/?view=spark-dotnet">.NET for Apache Spark API</a>'
```

**Comment by mairaw** [comment]
Left a suggestion for you on how to fix that one, but it's probably better to leave the other ones going to the API Browser page and not one specific namespace.

File: index.md [hunk 2]
```diff
@@ -247,7 +247,7 @@ ms.date: "12/04/2018"
                                                 <h3><a href="/dotnet/spark">.NET for Apache Spark Guide</a></h3>
                                                 <p><a href="/dotnet/spark/tutorials">Tutorials</a></p>
                                                 <p><a href="/dotnet/spark/resources">Resources</a></p>
-                                                <p><a href="/dotnet/api?view=spark-dotnet">.NET for Apache Spark API reference</a></p>
+                                                <p><a href="https://docs.microsoft.com/dotnet/api/?view=spark-dotnet">.NET for Apache Spark API reference</a></p>
```
Comment (mairaw) [inline index.md]: This one works as-is
```suggestion
                                                <p><a href="/dotnet/api/?view=spark-dotnet">.NET for Apache Spark API reference</a></p>
```

**Review by JasonWHowell — APPROVED** [review]
Links work from the TOC and the landing page as expected. Looks good to me! #sign-off

**Pull request merged**